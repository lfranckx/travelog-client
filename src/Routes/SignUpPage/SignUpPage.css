.sign-up-section {
    margin: 0 auto;
    text-align: center;
    display: block;
}

.sign-up-form label, .sign-up-form input {
    font-size: 20px;
}

/* First and Last Name */
.name-box {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.name-input {
    display: flex;
    flex-direction: column;
    margin: 5px;
}

.password-box {
    display: block;
    margin: 10px;
}

.sign-up-form button:active {
    transform: translateY(2px);
}

@media (max-width: 500px) {
    .name-input label, .name-input input {
        font-size: 14px;
    }

    .password-box label, .password-box input {
        font-size: 14px;
    }
}