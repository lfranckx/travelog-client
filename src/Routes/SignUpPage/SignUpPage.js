import React, { Component } from 'react';
import SignUpForm from '../../Components/SignUpForm/SignUpForm';
import './SignUpPage.css';

export default class SignUpPage extends Component {
    static defaultProps = {
        location: {},
        history: {
            push: () => {}
        }
    }

    handleSubmitSuccess = () => {
        const { location, history } = this.props;
        const destination = (location.state || {}).from || '/';
        history.push(destination);
    }

    render() {
        return (
            <>
                <section className="sign-up-section">
                    <h2>Sign Up</h2>
                    <SignUpForm 
                        onSubmitSuccess={this.handleSubmitSuccess}
                    />
                </section>
            </>
        );
    }
}