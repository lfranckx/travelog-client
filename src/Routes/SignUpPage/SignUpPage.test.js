import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignUpPage from './SignUpPage';

describe(`SignUpPage component`, () => {

    it(`renders sign up page without crashing`, () => {
        const wrapper = shallow(<SignUpPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})