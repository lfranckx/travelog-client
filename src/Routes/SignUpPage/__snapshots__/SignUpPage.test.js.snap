// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage component renders sign up page without crashing 1`] = `
<Fragment>
  <section
    className="sign-up-section"
  >
    <h2>
      Register
    </h2>
    <SignUpForm
      onSubmitSuccess={[Function]}
    />
  </section>
</Fragment>
`;
